### PR TITLE
ref(contextmenu): simplify the hook API

### PR DESCRIPTION
**Pull request by JonasBa**
Some of the logic was a bit all over the place, this centralizes all the required context menu functionality inside the hook so that components can just focus on the rendering part.

https://user-images.githubusercontent.com/9317857/170515913-922a1d95-6680-42f9-8dbb-6728bb754345.mp4

### CODE REVIEW

File: static/app/components/profiling/ProfilingContextMenu/profilingContextMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
+import {forwardRef} from 'react';
```
Comment (JonasBa) [inline static/app/components/profiling/ProfilingContextMenu/profilingContextMenu.tsx:1]: All of the components here were just moved

File: static/app/components/profiling/ProfilingContextMenu/profilingContextMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  box-shadow: ${p => p.theme.dropShadowHeavy};
+  width: auto;
+  overflow: auto;
+
+  &:focus {
+    outline: none;
+  }
+`;
+
+export {Menu as ProfilingContextMenu};
```
Comment (JonasBa) [inline static/app/components/profiling/ProfilingContextMenu/profilingContextMenu.tsx:32]: I renamed the exports so we dont pollute the global space anymore

File: static/app/components/profiling/flamegraphOptionsContextMenu.tsx [hunk 3]
```diff
@@ -205,173 +28,82 @@ const FLAMEGRAPH_VIEW_OPTIONS: FlamegraphViewOptions = ['top down', 'bottom up']
...
-    props.contextMenuCoordinates && containerCoordinates && menuCoordinates
-      ? computeBestContextMenuPosition(
-          props.contextMenuCoordinates,
-          containerCoordinates,
-          menuCoordinates
-        )
-      : null;
-
-  return (
+  return props.contextMenu.open ? (
```
Comment (JonasBa) [inline static/app/components/profiling/flamegraphOptionsContextMenu.tsx:38]: No more listeners and resizeobservers here, just rendering the menu

File: static/app/utils/profiling/hooks/useContextMenu.tsx [hunk 4]
```diff
@@ -39,66 +39,55 @@ export function useContextMenu({container}: UseContextMenuOptions) {
 
   // We wrap the setOpen function in a useEffect so that we also clear the keyboard
   // tabIndex when a menu is closed. This prevents tabIndex from being persisted between render
-  const wrapSetOpen = useCallback(
```
Comment (Zylphrex) [inline static/app/utils/profiling/hooks/useContextMenu.tsx]: Any reason you removed all the `useCallback` calls?

Comment (JonasBa) [reply static/app/utils/profiling/hooks/useContextMenu.tsx]: Added them back, I was testing to see if there was a stale closure and forgot to  add it back

File: static/app/components/profiling/ProfilingContextMenu/profilingContextMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+  checked?: boolean;
+}
+
+const MenuItemCheckbox = styled(
+  forwardRef(
+    (props: MenuItemCheckboxProps, ref: React.Ref<HTMLDivElement> | undefined) => {
+      const {children, checked, className, style, ...rest} = props;
+
+      return (
+        // @ts-ignore this ref is forwarded
```
Comment (Zylphrex) [inline static/app/components/profiling/ProfilingContextMenu/profilingContextMenu.tsx:45]: is this `@ts-ignore` needed?

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**